test/Elastic.Apm.Feature.Tests/ConfigurationSteps.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to Elasticsearch B.V under
+// one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+using TechTalk.SpecFlow;
+
+namespace Elastic.Apm.Feature.Tests
+{
+	[Binding]
+	public class ConfigurationSteps
+	{
+		private readonly ScenarioContext _scenarioContext;
+
+		public ConfigurationSteps(ScenarioContext scenarioContext) => _scenarioContext = scenarioContext;
+
+		[When("^service name is not set$")]
+		public void WhenServiceNameIsNotSet() { }
+
+		[When("^service version is not set$")]
+		public void WhenServiceVersionIsNotSet() { }
+
+		[When("^service name is set to '(.*?)'$")]
+		public void WhenServiceNameIsSetTo(string name)
+		{
+			var configuration = _scenarioContext.Get<TestConfiguration>();

```

Comment:
A bit sad we can't just use `Elastic.Apm.Tests.Utilities.MockConfiguration` here.

Reply:
MockConfiguration would require some work to factor it for usage here. I'm thinking

1. Remove the parsing functions used in MockConfiguration, and test them separately
2. Add public setters to the existing public properties
3. Change the constructor to be parameterless and initialize field members with their default values

I considered making those changes for this PR, but it felt like they would be large enough to open a separate PR for.